src/main/java/com/project/ita5/answer/AnswerServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+        List<AnswerPerson> personAnswerList = new ArrayList<>();
+        List<Person> personList = personRepository.findAll();
+        for (Person person : personList) {
+            List<Answer> answerList = answerRepository.findAllByPersonIdOrderByQuestionId(person.getId());
+            personAnswerList.add(new AnswerPerson(person, answerList));
+        }
+        return new ResponseEntity<List<AnswerPerson>>(personAnswerList, HttpStatus.OK);
+    }
+
+    @Override
+    public List<Answer> findAll() {
+        return answerRepository.findAll();
+    }
+
+    @Override
+    public Answer find(String id) {
+        return answerRepository.findById(id).orElse(null);
+    }
+
+
+    public ResponseEntity<AnswerPerson> findPersonInfo(String id) {
+        Person person = personRepository.findById(id).orElse(null);
+        if (person != null) {
+            List<Answer> answerList = answerRepository.findAllByPersonIdOrderByQuestionId(person.getId());
+            return new ResponseEntity<AnswerPerson>(new AnswerPerson(person, answerList), HttpStatus.OK);
+        }
+        return new ResponseEntity("No such person exists", HttpStatus.BAD_REQUEST);
+    }
+
+    @Override
+    public ResponseEntity saveAll(AnswerPerson answers) {
+        ResponseEntity<Person> currentPerson = personService.save(answers.person);
+        if (currentPerson.getStatusCode().isError()) {
+            return currentPerson;
+        }
+
+        String personId = currentPerson.getBody().getId();
+        for (Answer answer :
+                answers.answerList) {
+            answerRepository.save(new Answer(
+                    Long.toString(generateSequence.generateSequence(Answer.SEQUENCE_NAME)),
+                    answer.getQuestionId(),
+                    answer.getAnswer(),
+                    personId));
+        }
+        emailService.sendConfirmationEmail(currentPerson.getBody().getEmail());
+        return new ResponseEntity(
+                new AnswerPerson(currentPerson.getBody(), answerRepository.findAllByPersonIdOrderByQuestionId(personId)),
+                HttpStatus.OK
+        );
+
+    }
+
+    @Override
+    public ResponseEntity updatePerson(Person person) {
+        Person personToUpdate = personRepository.findById(person.getId()).orElse(null);
+        String statusToUpdate;
+        Person personToReturn;
+        String notesToUpdate;
+        String applicationValuationToUpdate;
+        String interviewValuationToUpdate;
+        String testValuationToUpdate;
+        if ((person.getExtra().getStatus() == null)) {
+            notesToUpdate = person.getExtra().getNotes();
+            applicationValuationToUpdate = person.getExtra().getApplicationValuation();
+            interviewValuationToUpdate = person.getExtra().getInterviewValuation();
+            testValuationToUpdate = person.getExtra().getTestValuation();
+            statusToUpdate = personToUpdate.getExtra().getStatus();
+        } else {
```

Comment:
You could create a separate function to create ApplicationExtra and pass either personToUpdate or person to it to avoid duplicate code.

Reply:
now fixed